src/MerchantCenter/MerchantStatuses.php

```diff
@@ -467,21 +480,11 @@ protected function sum_status_counts( array $validated_mc_statuses ): void {
 			$this->product_statuses['products'][ $wc_product_id ][ $status ] = 1 + ( $this->product_statuses['products'][ $wc_product_id ][ $status ] ?? 0 );
 
 			// Aggregate parent statuses for mc_status postmeta.
-			try {
-				$wc_parent_id = $product_helper->maybe_swap_for_parent_id( $wc_product_id );
-				if ( $wc_parent_id === $wc_product_id ) {
-					continue;
-				}
-				$this->product_statuses['parents'][ $wc_parent_id ][ $status ] = 1 + ( $this->product_statuses['parents'][ $wc_parent_id ][ $status ] ?? 0 );
-			} catch ( InvalidValue $e ) {
-
-				// Don't include invalid products (or their parents).
-				do_action(
-					'woocommerce_gla_debug_message',
-					sprintf( 'Merchant Center product ID %s not found in this WooCommerce store.', $wc_product_id ),
-					__METHOD__,
-				);
+			$wc_parent_id = intval( $this->product_data_lookup[ $wc_product_id ]['maybe_parent_id'] );
+			if ( $wc_parent_id === $wc_product_id ) {
+				continue;
 			}
```

Comment:
If we always store the parent ID here we can just check with `! $wc_parent_id`

One disadvantage of moving away from `maybe_swap_for_parent_id` is that we don't actually check if the parent product is valid. Which is faster because we don't need to load it, but doesn't protect us from "orphaned variations". In this code at most we'd get an invalid count which wouldn't be an issue. Do you foresee it being a problem anywhere further down the line?

Reply:
I don't think it should be an issue since they would actually be products in the Merchant Center (correctly or not), so counting them wouldn't be completely wrong.
I take it that orphaned variations come from manual DB changes or other hiccups?

Reply:
Yes, mostly manual DB changes or plugins doing it wrong. Although it's popular enough that WooCommerce added a tool to clear them up in WooCommerce > Status > Tools.